source/common/stats/thread_local_store.cc

```diff
@@ -277,55 +332,41 @@ StatType& ThreadLocalStoreImpl::ScopeImpl::safeMakeStat(
   // We must now look in the central store so we must be locked. We grab a reference to the
   // central store location. It might contain nothing. In this case, we allocate a new stat.
   Thread::LockGuard lock(parent_.lock_);
-  auto iter = central_cache_map.find(stat_key);
+  auto iter = central_cache_map.find(name);
   std::shared_ptr<StatType>* central_ref = nullptr;
   if (iter != central_cache_map.end()) {
     central_ref = &(iter->second);
   } else if (parent_.checkAndRememberRejection(name, central_rejected_stats, tls_rejected_stats)) {
     // Note that again we do the name-rejection lookup on the untruncated name.
     return null_stat;
   } else {
-    // If we had to truncate, warn now that we've missed all caches.
-    if (truncation_buffer != nullptr) {
-      ENVOY_LOG_MISC(
-          warn,
-          "Statistic '{}' is too long with {} characters, it will be truncated to {} characters",
-          name, name.size(), truncation_buffer->size());
-    }
-
-    std::vector<Tag> tags;
-
-    // Tag extraction occurs on the original, untruncated name so the extraction
-    // can complete properly, even if the tag values are partially truncated.
-    std::string tag_extracted_name = parent_.getTagsForName(name, tags);
+    TagExtraction extraction(parent_, name);
+    // std::shared_ptr<StatType> stat = make_stat(parent_.alloc_, extraction.truncatedStatName(),
```

Comment:
? delete?

Reply:
Ouch, yes. WDYT about a quick follow-up for that?

Reply:
never mind; deleted.